### PR TITLE
Support docker-py >= 2.0.0

### PR DESCRIPTION
**Pull request by berryp**
[Build Status](https://travis-ci.org/berryp/hyper_sh)

### Changes:

* Version 2 of `docker-py` contained breaking changes, which in turn broke this library when when installing after the v2 release. This change is to support versions >= 2.0.0.
* Pin `docker` requirement to >= 2.0.0.
* Updated tests. Travis builds pass.
* Added Travis build status badge.
* Uppercase environment variables for Hyper config are now supported.
* Extra config validity checks.
* Minor code cleanup, including `yapf` formatting.

### CODE REVIEW

File: hyper_sh/__init__.py [hunk 1]
```diff
@@ -1,58 +1,80 @@
...
-        else:
-            raise RuntimeError("unable to guess config from default file or environment")
-        return config
+            # We don't read the file here as we allow the user to
+            # pass the location to a config file, or a config object
+            # during initialisation.
+            return default_config_file
+
+        if {'HYPER_ENDPOINT', 'HYPER_ACCESSKEY', 'HYPER_SECRETKEY'}.issubset(os.environ):
+            return APIClient.config_object(os.environ['HYPER_ENDPOINT'], os.environ['HYPER_ACCESSKEY'],
```
Comment (tardyp) [inline hyper_sh/__init__.py]: Thanks for cleaning up my messy code.

File: tests/test_integration.py [hunk 2]
```diff
@@ -1,23 +1,23 @@
 from unittest.case import SkipTest
 
-from hyper_sh import Client
+from hyper_sh import APIClient
 
 
 def assertSetup():
     try:
-        Client.guess_config()
```
Comment (tardyp) [inline tests/test_integration.py:8]: wow. This breaks the API. We need to call it hyper_sh 2.0.0 then...

**Review by tardyp — APPROVED** [review]
I don't like to break API, but this makes sense if it is the same API as docker 2.0

File: tests/test_integration.py (hunk 2 above)
Comment (berryp) [reply tests/test_integration.py:8]: Yeah, that makes sense.

File: hyper_sh/__init__.py (hunk 1 above)
Comment (berryp) [reply hyper_sh/__init__.py]: 👍

**Comment by berryp** [comment]
Closing in favour of a cleaner history which includes further improvements and unit tests.

**Pull request closed without merging**